Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py

```diff
@@ -0,0 +1,370 @@
+    'initial-access': "Initial Access",
+    'discovery': "Discovery",
+    'execution': "Execution",
+    'installation': "Installation",
+    'delivery': "Delivery",
+    'weaponization': "Weaponization",
+    'act-on-objectives': "Actions on Objectives",
+    'command-and-control': "Command \u0026 Control"
+}
+
+RELATIONSHIP_TYPES = EntityRelationship.Relationships.RELATIONSHIPS_NAMES.keys()
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+
+class Client:
+
+    def __init__(self, url, proxies, verify, tags: list = None,
+                 tlp_color: Optional[str] = None):
+        self.base_url = url
+        self.proxies = proxies
+        self.verify = verify
+        self.tags = [] if not tags else tags
+        self.tlp_color = tlp_color
+        self.server: Server
+        self.api_root: List[ApiRoot]
+        self.collections: List[Collection]
+
+    def get_server(self):
+        server_url = urljoin(self.base_url, '/taxii/')
+        self.server = Server(server_url, verify=self.verify, proxies=self.proxies)
+
+    def get_roots(self):
+        self.api_root = self.server.api_roots[0]
+
+    def get_collections(self):
+        self.collections = [x for x in self.api_root.collections]  # type: ignore[attr-defined]
+
+    def initialise(self):
+        self.get_server()
+        self.get_roots()
+        self.get_collections()
+
+    def build_iterator(self, create_relationships=False, limit: int = -1) -> List:
+        """Retrieves all entries from the feed.
+
+        Returns:
+            A list of objects, containing the indicators.
+        """
+        indicators: List[Dict] = list()
+        mitre_id_list: Set[str] = set()
+        relationships_list = []
+        id_to_name: Dict = {}
+        counter = 0
+
+        # For each collection
+        for collection in self.collections:
+
+            # Stop when we have reached the limit defined
+            if 0 < limit <= counter:
+                break
+
+            # Establish TAXII2 Collection instance
+            collection_url = urljoin(self.base_url, f'stix/collections/{collection.id}/')
+            collection_data = Collection(collection_url, verify=self.verify, proxies=self.proxies)
+
+            # Supply the collection to TAXIICollection
+            tc_source = TAXIICollectionSource(collection_data)
+
+            filter_objs = {
+                "Technique": {"name": "attack-pattern", "filter": Filter("type", "=", "attack-pattern")},
+                "Mitigation": {"name": "course-of-action", "filter": Filter("type", "=", "course-of-action")},
+                "Group": {"name": "intrusion-set", "filter": Filter("type", "=", "intrusion-set")},
+                "Malware": {"name": "malware", "filter": Filter("type", "=", "malware")},
+                "Tool": {"name": "tool", "filter": Filter("type", "=", "tool")},
+                "relationships": {"name": "relationships", "filter": Filter("type", "=", "relationship")},
+            }
```

Comment:
consider moving this to a diffrent creator function because this code seems to be constant and just making the function longer.

Reply:
done